br/pkg/lightning/backend/local/region_job.go

```diff
@@ -0,0 +1,620 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/google/uuid"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/kvproto/pkg/errorpb"
+	sst "github.com/pingcap/kvproto/pkg/import_sstpb"
+	"github.com/pingcap/kvproto/pkg/kvrpcpb"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/tidb/br/pkg/lightning/common"
+	"github.com/pingcap/tidb/br/pkg/lightning/config"
+	"github.com/pingcap/tidb/br/pkg/lightning/log"
+	"github.com/pingcap/tidb/br/pkg/lightning/metric"
+	"github.com/pingcap/tidb/br/pkg/logutil"
+	"github.com/pingcap/tidb/br/pkg/membuf"
+	"github.com/pingcap/tidb/br/pkg/restore/split"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/mathutil"
+	"go.uber.org/zap"
+)
+
+type jobStageTp string
+
+const (
+	regionScanned jobStageTp = "regionScanned"
+	wrote         jobStageTp = "wrote"
+	ingested      jobStageTp = "ingested"
+	needRescan    jobStageTp = "needRescan"
+)
+
+func (j jobStageTp) String() string {
+	return string(j)
+}
+
+// regionJob is dedicated to import the data in [keyRange.start, keyRange.end) to a region.
+type regionJob struct {
+	keyRange Range
+	// TODO: check the keyRange so that it's always included in region
+	region *split.RegionInfo
+	// stage should be updated only by convertStageTo
+	stage jobStageTp
+
+	engine          *Engine
+	regionSplitSize int64
+	regionSplitKeys int64
```

Comment:
```suggestion
	maxRegionSize int64
	maxRegionKeys int64
```

Reply:
These names correspond to config item, I prefer we don't change them